### PR TITLE
Add clap

### PR DESCRIPTION
**Pull request by balroggg**
Add parsing cli arguments by `clap` crate.

Add parsing some vars from config:
- twitch channel
- tick delay
- max messages
- date shown
- maximum username length
- username alignment
- palette

Example:
```
Running `target/debug/twt -h`
twitch-tui 1.4.1
Xithrius
Twitch chat in the terminal.

USAGE:
    twt [OPTIONS]

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -c, --channel <CHANNEL>                         The streamer's name
    -s, --date-shown <true/false>                   If the time and date is to be shown
    -m, --max-messages <MESSAGES>                   The maximum amount of messages to be stored
    -u, --maximum-username-length <LENGTH>          The longest a username can be
    -p, --palette <PALETTE>
            The color palette for the username column: pastel (default), vibrant, warm, cool

    -t, --tick-delay <DELAY>                        The delay in milliseconds between terminal updates
    -a, --username-alignment <left/center/right>    Side the username should be aligned to
```

Fix issue with text alignment. Now column with `Username` crop without error(maybe add some minimal value of crop, length of string `Username` or max value of any username in chat).

### CODE REVIEW

**Comment by Xithrius** [comment]
I'll be reviewing this soon. I've closed the other CLI parsing PR.

**Comment by balroggg** [comment]
Fix test `test_text_align_center`. Please check if I break logic, I try to use more core rust features.

**Comment by balroggg** [comment]
> I'll be reviewing this soon. I've closed the other CLI parsing PR.

Ok, thanks.

File: src/main.rs [hunk 1]
```diff
@@ -13,8 +15,51 @@ mod utils;
...
 #[tokio::main]
 async fn main() -> Result<()> {
+    let arg_matches = clap::App::new("twitch-tui")
+        .version("1.4.1")
+        .author("Xithrius")
+        .about("Twitch chat in the terminal.")
+        .args_from_usage("-c, --channel=[CHANNEL] 'The streamer's name'
+                          -t, --tick-delay=[DELAY] 'The delay in milliseconds between terminal updates'
+                          -m, --max-messages=[MESSAGES] 'The maximum amount of messages to be stored'
+                          -s, --date-shown=[true/false] 'If the time and date is to be shown'
```
Comment (Xithrius) [inline src/main.rs:25]:
```suggestion
                          -s, --date-shown=[true/false] 'If the time and date is to be shown (defaults to true)'
```

File: src/main.rs [hunk 2]
```diff
@@ -13,8 +15,51 @@ mod utils;
...
+                          -u, --maximum-username-length=[LENGTH] 'The longest a username can be'
+                          -a, --username-alignment=[left/center/right] 'Side the username should be aligned to'
+                          -p, --palette=[PALETTE] 'The color palette for the username column: pastel (default), vibrant, warm, cool'")
+        .get_matches();
+
     match CompleteConfig::new() {
-        Ok(config) => {
+        Ok(mut config) => {
+            // twitch section of config
```
Comment (Xithrius) [inline src/main.rs:33]:
```suggestion
            // Twitch section of the config
```

File: src/main.rs [hunk 3]
```diff
@@ -13,8 +15,51 @@ mod utils;
...
+            if let Some(ch) = arg_matches.value_of("channel") {
+                config.twitch.channel = ch.to_string();
+            }
+            // terminal section of config
```
Comment (Xithrius) [inline src/main.rs:37]:
```suggestion
            // Terminal section of the config
```

File: src/main.rs [hunk 4]
```diff
@@ -13,8 +15,51 @@ mod utils;
...
+            if let Some(tick_delay) = arg_matches.value_of("tick-delay") {
+                config.terminal.tick_delay = tick_delay.parse().unwrap();
+            }
+            if let Some(max_messages) = arg_matches.value_of("maximum-messages") {
+                config.terminal.maximum_messages = max_messages.parse().unwrap();
+            }
+            // frontend section of config
```
Comment (Xithrius) [inline src/main.rs:44]:
```suggestion
            // Frontend section of the config
```

File: src/main.rs [hunk 5]
```diff
@@ -13,8 +15,51 @@ mod utils;
...
 async fn main() -> Result<()> {
+    let arg_matches = clap::App::new("twitch-tui")
+        .version("1.4.1")
+        .author("Xithrius")
+        .about("Twitch chat in the terminal.")
+        .args_from_usage("-c, --channel=[CHANNEL] 'The streamer's name'
+                          -t, --tick-delay=[DELAY] 'The delay in milliseconds between terminal updates'
+                          -m, --max-messages=[MESSAGES] 'The maximum amount of messages to be stored'
+                          -s, --date-shown=[true/false] 'If the time and date is to be shown'
+                          -u, --maximum-username-length=[LENGTH] 'The longest a username can be'
```
Comment (Xithrius) [inline src/main.rs:26]:
```suggestion
                          -u, --maximum-username-length=[LENGTH] 'Maximum length for Twitch usernames'
```

File: src/main.rs [hunk 6]
```diff
@@ -1,7 +1,9 @@
+extern crate clap;
+
```
Comment (Xithrius) [inline src/main.rs:2]: This isn't needed.
```suggestion
```

**Review by Xithrius — CHANGES_REQUESTED** [review]

**Comment by balroggg** [comment]
Fix typo and format code in utils

File: src/main.rs [hunk 7]
```diff
@@ -13,8 +13,51 @@ mod utils;
...
 async fn main() -> Result<()> {
+    let arg_matches = clap::App::new("twitch-tui")
+        .version("1.4.1")
+        .author("Xithrius")
+        .about("Twitch chat in the terminal.")
+        .args_from_usage("-c, --channel=[CHANNEL] 'The streamer's name'
+                          -t, --tick-delay=[DELAY] 'The delay in milliseconds between terminal updates'
+                          -m, --max-messages=[MESSAGES] 'The maximum amount of messages to be stored'
+                          -s, --date-shown=[true/false] 'If the time and date is to be shown (defaults to true)'
+                          -u, --maximum-username-length=[LENGTH] 'Maximum length for Twitch usernames'
```
Comment (Xithrius) [inline src/main.rs:24]:
```suggestion
                          -u, --max-username-length=[LENGTH] 'Maximum length for Twitch usernames'
```

File: src/main.rs [hunk 8]
```diff
@@ -13,8 +13,51 @@ mod utils;
...
+        Ok(mut config) => {
+            // Twitch section of the config
+            if let Some(ch) = arg_matches.value_of("channel") {
+                config.twitch.channel = ch.to_string();
+            }
+            // Terminal section of the config
+            if let Some(tick_delay) = arg_matches.value_of("tick-delay") {
+                config.terminal.tick_delay = tick_delay.parse().unwrap();
+            }
+            if let Some(max_messages) = arg_matches.value_of("maximum-messages") {
```
Comment (Xithrius) [inline src/main.rs:39]:
```suggestion
            if let Some(max_messages) = arg_matches.value_of("max-messages") {
```

File: src/main.rs [hunk 9]
```diff
@@ -13,8 +13,51 @@ mod utils;
...
+                config.terminal.maximum_messages = max_messages.parse().unwrap();
+            }
+            // Frontend section of the config
+            if let Some(date_shown) = arg_matches.value_of("date-shown") {
+                config.frontend.date_shown = date_shown.parse().unwrap();
+            }
+            if let Some(maximum_username_length) = arg_matches.value_of("maximum-username-length") {
```
Comment (Xithrius) [inline src/main.rs:46]:
```suggestion
            if let Some(max_username_length) = arg_matches.value_of("max-username-length") {
```

**Review by Xithrius — CHANGES_REQUESTED** [review]
Just want the user to type less. I'll change the config vars later to match. 

Besides these things, everything else looks good.

After this, the PR shall be merged.

**Comment by balroggg** [comment]
> Just want the user to type less. I'll change the config vars later to match.
> 
> Besides these things, everything else looks good.
> 
> After this, the PR shall be merged.

Ok, good point.

**Review by Xithrius — APPROVED** [review]
Looks good. Thanks for the PR!

**Comment by Xithrius** [comment]
@balroggg I'll be releasing a new version of the TUI relatively soon, just got to do a few things and then we'll be good to go.

**Pull request merged**